raiden/tests/integration/long_running/test_settlement.py

```diff
@@ -189,6 +190,137 @@ def test_settle_is_automatically_called(
+        token_address,
+        target=TargetAddress(app0.address),
+        amount=PaymentAmount(10),
+        identifier=payment_id,
+        secret=secret,
+        secrethash=secrethash,
+    )
+
+    waiting.wait_for_received_transfer_result(app0, payment_id, PaymentAmount(10), 1.0, secrethash)
+
+    RaidenAPI(app0).channel_close(registry_address, token_address, app1.address, coop_settle=True)
+
+    channel_state = views.get_channelstate_for(
+        views.state_from_raiden(app0), registry_address, token_address, app1.address
+    )
+    assert channel_state is None
+
+    token_network = views.get_token_network_by_address(
+        views.state_from_raiden(app0), token_network_address
+    )
+    assert token_network
+
+    assert (
+        channel_identifier
+        not in token_network.partneraddresses_to_channelidentifiers[app1.address]
+    )
+
+    assert app0.wal, MSG_BLOCKCHAIN_EVENTS
+    assert app0.alarm, MSG_BLOCKCHAIN_EVENTS
+    state_changes = app0.wal.storage.get_statechanges_by_range(RANGE_ALL_STATE_CHANGES)
+
+    for app, partner_app in ((app0, app1), (app1, app0)):
+        assert app.wal, MSG_BLOCKCHAIN_EVENTS
+        assert app.alarm, MSG_BLOCKCHAIN_EVENTS
+        state_changes = app.wal.storage.get_statechanges_by_range(RANGE_ALL_STATE_CHANGES)
+
+        assert not search_for_item(
+            state_changes,
+            ContractReceiveChannelClosed,
+            {
+                "token_network_address": token_network_address,
+                "channel_identifier": channel_identifier,
+            },
+        )
+
+        assert search_for_item(
+            state_changes,
+            ContractReceiveChannelWithdraw,
+            {
+                "token_network_address": token_network_address,
+                "channel_identifier": channel_identifier,
+            },
+        )
+        assert search_for_item(
+            state_changes,
+            ContractReceiveChannelSettled,
+            {
+                "token_network_address": token_network_address,
+                "channel_identifier": channel_identifier,
+                "our_transferred_amount": app_to_expected_balances[app],
+                "partner_transferred_amount": app_to_expected_balances[partner_app],
+            },
+        )
+
+
+@raise_on_failure
+@pytest.mark.parametrize("number_of_nodes", [2])
+def test_coop_settle_fails_with_pending_lock(
+    raiden_network: List[RaidenService], token_addresses: List[TokenAddress]
+) -> None:
+    """Settle is automatically called by one of the nodes."""
+    app0, app1 = raiden_network
+    registry_address = app0.default_registry.address
+    token_address = token_addresses[0]
+
+    RaidenAPI(app1).transfer_and_wait(
+        app0.default_registry.address,
+        token_address,
+        target=TargetAddress(app0.address),
+        amount=PaymentAmount(10),
+    )
+
+    RaidenAPI(app0).channel_close(registry_address, token_address, app1.address, coop_settle=True)
```

Comment:
It might be the case that locally the lock is always pending still, but this is not guaranteed. It depends in which order the greenlets are executed that the lock might not be pending anymore. Especially in the CI we usually see different behavior. 

IOW, to be really sure that the payment is still pending we must make sure that the incoming message of the unlock is not processed before `channel_close())` is called.